wireclient/wireclient.go

```diff
@@ -487,21 +484,16 @@ func (c *Conn) loginScramSHA256(ctx context.Context, username, password, authDB
 		"$db", authDB,
 	)
 
-	// one `saslStart`, two `saslContinue` requests
+	// one `saslStart`, two `saslContinue` for servers that don't support `skipEmptyExchange`
 	steps := 3
 
-	if skipEmptyExchange {
-		if err = cmd.Add("options", wirebson.MustDocument("skipEmptyExchange", true)); err != nil {
-			return fmt.Errorf("wireclient.Conn.loginScramSHA256: %w", err)
-		}
-
-		// only one `saslContinue`
-		steps = 2
+	if err = cmd.Add("options", wirebson.MustDocument("skipEmptyExchange", true)); err != nil {
+		return fmt.Errorf("wireclient.Conn.loginScramSHA256: %w", err)
 	}
 
 	for step := 1; step <= steps; step++ {
```

Comment:
```suggestion
	// One `saslStart`, two `saslContinue` for servers that don't support `skipEmptyExchange`;
	// one and one for those who do.
	for step := 1; step <= 3; step++ {
```

Or something like that. No need to have a variable for something that is not variable.